tests/test_class_db.h

```diff
@@ -31,12 +31,800 @@
+		if (builtin_types.find(p_type_ref.name) >= 0) {
+			return true;
+		}
+
+		if (p_type_ref.is_enum) {
+			if (enum_types.find(p_type_ref.name) >= 0) {
+				return true;
+			}
+
+			// Enum not found. Most likely because none of its constants were bound, so it's empty. That's fine. Use int instead.
+			return builtin_types.find(names_cache.int_type);
+		}
+
+		return false;
+	}
+};
+
+bool arg_default_value_is_assignable_to_type(const Context &p_context, const Variant &p_val, const TypeReference &p_arg_type) {
+	if (p_arg_type.name == p_context.names_cache.variant_type) {
+		// Variant can take anything
+		return true;
+	}
+
+	switch (p_val.get_type()) {
+		case Variant::NIL:
+			return p_context.find_exposed_class(p_arg_type) ||
+				   p_context.names_cache.is_nullable_type(p_arg_type.name);
+		case Variant::BOOL:
+			return p_arg_type.name == p_context.names_cache.bool_type;
+		case Variant::INT:
+			return p_arg_type.name == p_context.names_cache.int_type ||
+				   p_arg_type.name == p_context.names_cache.float_type ||
+				   p_arg_type.is_enum;
+		case Variant::FLOAT:
+			return p_arg_type.name == p_context.names_cache.float_type;
+		case Variant::STRING:
+		case Variant::STRING_NAME:
+			return p_arg_type.name == p_context.names_cache.string_type ||
+				   p_arg_type.name == p_context.names_cache.string_name_type ||
+				   p_arg_type.name == p_context.names_cache.node_path_type;
+		case Variant::NODE_PATH:
+			return p_arg_type.name == p_context.names_cache.node_path_type;
+		case Variant::TRANSFORM:
+		case Variant::TRANSFORM2D:
+		case Variant::BASIS:
+		case Variant::QUAT:
+		case Variant::PLANE:
+		case Variant::AABB:
+		case Variant::COLOR:
+		case Variant::VECTOR2:
+		case Variant::RECT2:
+		case Variant::VECTOR3:
+		case Variant::_RID:
+		case Variant::ARRAY:
+		case Variant::DICTIONARY:
+		case Variant::PACKED_BYTE_ARRAY:
+		case Variant::PACKED_INT32_ARRAY:
+		case Variant::PACKED_INT64_ARRAY:
+		case Variant::PACKED_FLOAT32_ARRAY:
+		case Variant::PACKED_FLOAT64_ARRAY:
+		case Variant::PACKED_STRING_ARRAY:
+		case Variant::PACKED_VECTOR2_ARRAY:
+		case Variant::PACKED_VECTOR3_ARRAY:
+		case Variant::PACKED_COLOR_ARRAY:
+		case Variant::CALLABLE:
+		case Variant::SIGNAL:
+			return p_arg_type.name == Variant::get_type_name(p_val.get_type());
+		case Variant::OBJECT:
+			return p_context.find_exposed_class(p_arg_type);
+		case Variant::VECTOR2I:
+			return p_arg_type.name == p_context.names_cache.vector2_type ||
+				   p_arg_type.name == Variant::get_type_name(p_val.get_type());
+		case Variant::RECT2I:
+			return p_arg_type.name == p_context.names_cache.rect2_type ||
+				   p_arg_type.name == Variant::get_type_name(p_val.get_type());
+		case Variant::VECTOR3I:
+			return p_arg_type.name == p_context.names_cache.vector3_type ||
+				   p_arg_type.name == Variant::get_type_name(p_val.get_type());
+		default:
+			ERR_PRINT("Unexpected Variant type: " + itos(p_val.get_type()));
```

Comment:
Use doc tests's INFO or FAIL?

Reply:
I think this could be moved outside the method call, and `CHECK_MESSAGE` accordingly. `FAIL` fails the entire test suite, but we're interested in checking more assertions...

Reply:
Or those individual equality checks could be replaced with `CHECK_MESSAGE`, so you know exactly what failed.

Reply:
You messaged that you resolved the other issues.

Reply:
Yeah I have yet to resolve this. 😛

Reply:
Yeah I overthink this, `FAIL` is good because that would denote we have a `Variant` type which is not handled yet (new type or obsolete), so this would immediately notify testers that the checks above must be updated accordingly.